### ISSUE TITLE
Missing freetype library on Debian Jessie

### ISSUE THREAD

**Issue by egraether** [probonopd/linuxdeployqt#430]
[Sourcetrail](https://github.com/CoatiSoftware/Sourcetrail/) recently started offering an [AppImage](https://github.com/CoatiSoftware/Sourcetrail/releases/tag/2020.1.117).

One user experienced this error when using the AppImage on Debian Jessie ([source](https://github.com/CoatiSoftware/Sourcetrail/issues/869#issuecomment-604264298)):
`x86_64.AppImage: symbol lookup error: /tmp/.mount_SourcecUdaPg/usr/plugins/platforms/../../lib/libQt5XcbQpa.so.5: undefined symbol: FT_Get_Font_Format`

[According to StackExchange](https://superuser.com/questions/1451321/qt-startup-error-symbol-lookup-error-libqt5xcbqpa-so-5-undefined-symbol-ft-g) this can be fixed by installing the library `freetype-devel`. We built the AppImage inside a [Docker container](https://github.com/CoatiSoftware/Sourcetrail/blob/2020.1.117/deployment/dockerfiles/linux/centos7_64_base/Dockerfile) running CentOS 7, and the library `freetype-devel` is installed there.

Why is the library not present in the final AppImage?
Do we have to manually add it?
Is simply adding it to the `lib` directory sufficient?

**Comment by probonopd** [comment]
According to https://superuser.com/a/1532307, freetype renamed a symbol, thus breaking binary compatibility.

> Programs that were compiled with freetype >= 2.6 may use the new API and, by doing this, such programs will not work with older versions of freetype.

The contents of the AppImage should be compiled on an older build system with freetype < 2.6.

**Comment by eresonance** [comment]
Have this same issue on a RHEL7 box that requires an older version of freetype to support an older tool we use. Unfortunately the AppImage doesn't seem to include the newer freetype lib, at least that's what I'm assuming is the issue here.

**Comment by probonopd** [comment]
Indeed, we are assuming that those libs are part of each target distribution in a recent enough version:

https://github.com/AppImage/pkg2appimage/blob/678e5e14122f14a12c54847213585ea803e1f0e1/excludelist#L105-L109

Solution: Build what goes into the AppImage on a distribution that is no newer than all the target systems.
